realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -250,73 +236,37 @@ private static synchronized Realm create(RealmConfiguration configuration) {
                 migrateRealm(configuration);
             }
 
-            return createAndValidate(configuration, true, autoRefresh);
+            return createAndValidate(configuration, columnIndices);
```

Comment:
Before entering migration block, there should be no opened typed Realm, which means, columnIndices should always be null before migration.